# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Zainab-Aijaz/WEEK1_ML_Assignment_FlyRank_Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:

# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


*I'm working the Refresh / Content Opportunity Scoring lane: which pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring. I chose it because it ends in something a real person can act on — a ranked list with a reason attached — rather than just a descriptive pattern. It also has the clearest cost of being wrong (a content team spends real hours on the wrong page), which makes it easier to write honest, checkable claims later.*

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

*Decision it improves: which pages an SEO/content team works on next, and what they do to each one (refresh, expand, protect, prune, or just watch). Who acts on it: a content strategist or SEO analyst working through a review queue, prioritizing limited hours across potentially thousands of pages. Cost of a wrong call: two kinds. A false positive burns real hours refreshing a page that wouldn't have decayed anyway (opportunity cost — that time didn't go to a page that actually needed it). A false negative lets a page that's already leaking traffic keep leaking while it sits unflagged, and by the time anyone notices, the recovery is slower and more expensive than an early catch would've been*

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
else:
    # find the repo root from wherever this kernel started
    while not os.path.isdir("data/raw") and os.getcwd() != "/":
        os.chdir("..")

print("Working dir:", os.getcwd())
assert os.path.exists("data/raw/content_refresh_anonymized.csv"), "starter CSV not found — are you at the repo root?"
print("Starter data found. You're ready.")

Working dir: /content/flyrank-ml-internship-starter
Starter data found. You're ready.


In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Pages:", len(df))
print("Clients:", df["client_id"].nunique())

declining_share = (df["trend_direction"] == "down").mean()
print(f"Pages trending down (last 30d vs prev 30d): {declining_share:.1%}")

zero_click_share = (df["clicks_last_30d"] == 0).mean()
print(f"Pages with zero clicks in the last 30 days: {zero_click_share:.1%}")

decay_risk = (df["avg_position"] <= 10) & (df["trend_direction"] == "down")
print(f"Page-one pages (avg position <=10) that are declining: {decay_risk.mean():.1%}")

Pages: 30000
Clients: 32
Pages trending down (last 30d vs prev 30d): 54.2%
Pages with zero clicks in the last 30 days: 61.2%
Page-one pages (avg position <=10) that are declining: 24.4%


*Across 30,000 pages from 32 clients: 54.2% are trending down in clicks over the last 30 days vs. the prior 30. 61.2% got zero clicks in the last 30 days — a huge pool nobody would think to manually check one by one. Most strikingly, 24.4% of page-one pages (avg position ≤10) are actively declining — these are the highest-value pages to lose quietly, since they're already winning real estate and could slip off page one before anyone notices. That combination — high volume, mostly-invisible signal, real cost if missed — is exactly the "worth 7 weeks" case for this lane.*

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

*What this work will be able to say: which pages look stale, thin, or declining based on observed signals; a ranked, directional priority order for review; a plausible reason attached to each flagged page; a decision-support tool that narrows a huge page list down to a short one worth a human's time.*
*What it will never say: that a refresh will cause a ranking or traffic increase (correlation in the training window isn't proof of a causal effect); that it's reverse-engineered or predicted Google's actual ranking algorithm; that a "decline" flag is a guarantee anything is wrong — it's a signal worth a look, not a verdict. Every output stays labeled observed/directional/decision-support, never causal proof.*

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.